pkg/logcli/client/client.go

```diff
@@ -237,6 +260,27 @@ func (c *DefaultClient) wsConnect(path, query string, quiet bool) (*websocket.Co
 		h.Set("X-Scope-OrgID", c.OrgID)
 	}
 
+	if (c.Username != "" || c.Password != "") && (len(c.BearerToken) > 0 || len(c.BearerTokenFile) > 0) {
```

Comment:
You could instead assign precedence to these settings. I think it would make sense to have it fall back in order:
- bearer token
- bearer token file
- user name and password

Reply:
I think I would have to touch a lot more code for that? And if a user would set two or all three variants, he would not know in the end what would be taken. I did it essentially the same way as Prometheus does it: https://github.com/prometheus/common/blob/f45215cb5711443fffd211cab7671b268245c4a1/config/http_config.go#L197